salt/metalk8s/node/grains.sls

```diff
@@ -0,0 +1,9 @@
+Set control_plane_ip grain:
+  grains.present:
+    - name: metalk8s:control_plane_ip
+    - value: {{ salt['network.ip_addrs'](cidr=salt['pillar.get']('networks:control_plane'))[0] }}
+
+Set workload_plane_ip grain:
+  grains.present:
+    - name: metalk8s:workload_plane_ip
+    - value: {{ salt['network.ip_addrs'](cidr=salt['pillar.get']('networks:workload_plane'))[0] }}
```

Comment:
What if we fail to find IPs in the configured CIDRs? Maybe the user misconfigured these networks... Maybe we could reuse the `if/else` structure here, so that when deploying a new node one quickly identifies which IP was not found.

(This "not finding IPs" issue occured quite frequently in the CI, since we have two private networks, the DHCP request we do during provisioning in Terraform may fail, and we may only have one of the two IPs available)

Reply:
Done